src/main/java/com/czertainly/core/service/impl/LocalAdminServiceImpl.java

```diff
@@ -1,169 +1,49 @@
-        UserRequestDto requestDto = new UserRequestDto();
-        requestDto.setDescription(request.getDescription());
-        requestDto.setEmail(request.getEmail());
-        requestDto.setEnabled(request.getEnabled());
-        requestDto.setUsername(request.getUsername());
-        requestDto.setFirstName(request.getFirstName());
-        requestDto.setLastName(request.getLastName());
-        requestDto.setCertificateFingerprint(fingerprint);
-        return userManagementApiClient.createUser(requestDto);
-    }
-
-    private UserDetailDto updateUser(AddUserRequestDto request, String fingerprint, String certificateUuid, String userUuid) {
-        UserUpdateRequestDto updateRequestDto = new UserUpdateRequestDto();
-        updateRequestDto.setDescription(request.getDescription());
-        updateRequestDto.setEmail(request.getEmail());
-        updateRequestDto.setFirstName(request.getFirstName());
-        updateRequestDto.setLastName(request.getLastName());
-        updateRequestDto.setCertificateFingerprint(fingerprint);
-        updateRequestDto.setCertificateUuid(certificateUuid);
-        return userManagementApiClient.updateUser(userUuid, updateRequestDto);
-    }
-
-    private String getCertificateFingerprint(X509Certificate certificate) throws CertificateEncodingException, NoSuchAlgorithmException {
-        return CertificateUtil.getThumbprint(certificate.getEncoded());
-    }
+        String superadminRoleUuid = getSuperadminRoleUuid();
+        userDetailDto = userManagementService.updateRole(userDetailDto.getUuid(), superadminRoleUuid);
 
-    private Certificate uploadCertificate(String certificateUuid, String certificateData) throws CertificateException, NotFoundException {
-        Certificate certificate;
-        if (StringUtils.isNotBlank(certificateUuid)) {
-            certificate = certificateService.getCertificateEntity(SecuredUUID.fromString(certificateUuid));
-        } else {
-            X509Certificate x509Cert = CertificateUtil.parseCertificate(certificateData);
-            try {
-                certificate = certificateService.getCertificateEntityByFingerprint(CertificateUtil.getThumbprint(x509Cert));
-            } catch (NotFoundException | NoSuchAlgorithmException e) {
-                certificate = certificateService.createCertificateEntity(x509Cert);
-                certificateService.updateCertificateEntity(certificate);
-            }
-        }
-        return certificate;
+        return userDetailDto;
     }
 
-    private String getSuperAdminRoleUuid(){
-        return roleManagementApiClient.getRoles().getData().stream().filter(e -> e.getSystemRole().equals(true) && e.getName().equals("superadmin")).collect(Collectors.toList()).get(0).getUuid();
+    private String getSuperadminRoleUuid() {
+        return roleManagementApiClient.getRoles().getData().stream().filter(e -> e.getSystemRole().equals(true) && e.getName().equals(AuthHelper.SUPERADMIN_USERNAME)).toList().getFirst().getUuid();
```

Comment:
@lubomirw Why we are looking for `SUPERADMIN_USERNAME` that is not mandatory, when looking for the `superadmin` role?

Reply:
System roles/users share the same name. Name of constant is misleading but name refers to role name and also to user username. Constants have _USERNAME suffix since they are almost always used to refer to and retrieve users. Should it be renamed to avoid confusion?